Runtime/Microsoft.Dynamic/Utils/MathUtils.cs

```diff
@@ -1029,34 +1028,35 @@ out result
             return BigInt.Pow(self, exp);
         }
 
-        private static readonly BigInt intMaxValueAsBigInteger = new BigInt(int.MaxValue);
         public static BigInt Power(this BigInt self, long exp) {
-            BigInt expAsBigInt = new BigInt(exp);
-
-            if (self.IsZero) {
-                return BigInt.Zero;
-            } else if (expAsBigInt.IsZero || self.IsOne) {
-                return BigInt.One;
-            } else if (expAsBigInt.IsOne) {
-                return self;
-            } else if (exp < 0) {
+            if (exp < 0) {
                 throw new ArgumentOutOfRangeException("exponent", exp, "Must be at least 0");
```

Comment:
Not your mistake, but while you're in there...

First argument should be `"exp"` or `nameof(exp)`. Also, if we want it to still compile for WP75, it should be `throw ExceptionUtils.MakeArgumentOutOfRangeException` instead of `throw new ArgumentOutOfRangeException`.